packages/@aws-cdk/aws-batch/test/test.batch-job-queue.ts

```diff
@@ -0,0 +1,86 @@
+
+                // WHEN
+                new batch.JobQueue(stack, 'job-queue', {
+                    jobQueueName: 'test-job-queue',
+                    computeEnvironmentOrder: [
+                        {
+                            computeEnvironment: new batch.ComputeEnvironment(stack, 'compute-env', {
+                                computeResources: {
+                                    subnets: vpc.privateSubnets,
+                                    instanceRole: new iam.Role(stack, 'compute-role', {
+                                        assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
+                                    }),
+                                },
+                            }),
+                            order: 1,
+                        },
+                    ],
+                    priority: 1,
+                    state: JobQueueState.DISABLED,
+                });
+
+                // THEN
+                expect(stack).to(haveResourceLike('AWS::Batch::JobQueue', {
+                    JobQueueName: 'test-job-queue',
+                    ComputeEnvironmentOrder: [
+                        {
+                            Order: 1,
+                        },
+                    ],
+                    Priority: 1,
+                    State: 'DISABLED',
+                }, ResourcePart.Properties));
+
+                expect(stack).to(haveResource('AWS::Batch::ComputeEnvironment'));
+
+                test.done();
+            },
+            'should be able to get an existing queue by Arn'(test: Test) {
+                // GIVEN
+                const stack = new cdk.Stack();
+                const vpc = new ec2.Vpc(stack, 'compute-vpc');
+
+                // WHEN
+                const existingJobQueue = new batch.JobQueue(stack, 'job-queue', {
+                    computeEnvironmentOrder: [
+                        {
+                            computeEnvironment: new batch.ComputeEnvironment(stack, 'compute-env', {
+                                computeResources: {
+                                    subnets: vpc.privateSubnets,
+                                    instanceRole: new iam.Role(stack, 'compute-role', {
+                                        assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
+                                    }),
+                                },
+                            }),
+                            order: 1,
+                        },
+                    ],
+                    priority: 1,
+                    state: JobQueueState.DISABLED,
+                });
+
+                const jobQueueByArn = batch.JobQueue.fromJobQueueArn(stack, 'existing-job-queue', existingJobQueue.jobQueueArn);
+
+                // THEN
+                test.equal(jobQueueByArn.jobQueueArn, existingJobQueue.jobQueueArn);
+
+                test.done();
+            },
+        },
+    }
+};
```

Comment:
Actually, since this is adding the first L2 code, can you use Jest instead of NodeUnit? You have examples in the repo of other packages using Jest (like [IAM](https://github.com/aws/aws-cdk/pull/3890/files)).

Reply:
Jest would be perfect! I am much more comfortable testing using that.